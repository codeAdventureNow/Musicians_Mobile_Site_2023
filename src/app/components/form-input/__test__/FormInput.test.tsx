import {
  render,
  screen,
  fireEvent,
  waitFor,
  act,
  getByTestId,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import FormInput from '../form-input';
import renderer from 'react-test-renderer';

describe('Form Input', () => {
  const user = userEvent.setup();
  it('should render the Form', () => {
    const formInput = renderer.create(<FormInput />).toJSON();
    expect(formInput).toMatchSnapshot();
  });

  it('call the onSubmit function', async () => {
    const mockOnSubmit = jest.fn();
    const { getByTestId, getByRole } = render(
      <FormInput onSubmit={mockOnSubmit} />
    );

    await act(async () => {
      fireEvent.change(getByTestId('fullName'), {
        target: { value: 'Runberto Machna' },
      });
      fireEvent.change(getByTestId('zipCode'), {
        target: { value: 90210 },
      });
      fireEvent.change(getByTestId('email'), {
        target: { value: 'runberto@gmail.com' },
      });
      fireEvent.change(getByTestId('phoneNumber'), {
        target: { value: 4087067554 },
      });
      fireEvent.change(getByTestId('guitar'), {
        target: { value: ['guitar'] },
      });
      fireEvent.change(getByTestId('Referral'), {
        target: { value: 'Referral' },
      });
    });

    await act(async () => {
      fireEvent.click(getByRole('button'));
    });

    expect(mockOnSubmit).toHaveBeenCalled();
  });
});
