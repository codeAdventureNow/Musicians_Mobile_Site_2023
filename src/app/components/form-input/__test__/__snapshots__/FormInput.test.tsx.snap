// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Input should render the Form 1`] = `
<DocumentFragment>
  <div>
    <form
      class="formInput"
      data-testid="contact-form"
    >
      <label
        class="label_hidden"
      >
        Date
      </label>
      <input
        class="label_hidden"
        name="date"
        type="text"
      />
      <label
        class="label"
        for="fullName"
      >
         First/Last Name* 
      </label>
      <input
        class="input"
        data-testid="fullName"
        id="fullName"
        name="fullName"
        type="text"
      />
      <label
        class="label"
        for="zipCode"
      >
         Zip Code* 
      </label>
      <input
        class="input"
        data-testid="zipCode"
        id="zipCode"
        name="zipCode"
        type="number"
      />
      <label
        class="label"
        for="email"
      >
         Email* 
      </label>
      <input
        class="input"
        data-testid="email"
        id="email"
        name="email"
        type="email"
      />
      <label
        class="label"
        for="phoneNumber"
      >
         Phone* 
      </label>
      <input
        class="input"
        data-testid="phoneNumber"
        id="phoneNumber"
        name="phone"
        type="number"
      />
      <label
        class="label"
      >
         Which musical instruments would you like to learn?* 
      </label>
      <div
        class="flex"
      >
        <div
          class="checkbox"
        >
          <label
            class="label"
            for="piano"
          >
            piano
          </label>
          <input
            class="input"
            data-testid="piano"
            id="piano"
            name="instruments"
            type="checkbox"
            value="piano"
          />
        </div>
        <div
          class="checkbox"
        >
          <label
            class="label"
            for="guitar"
          >
            guitar
          </label>
          <input
            class="input"
            data-testid="guitar"
            id="guitar"
            name="instruments"
            type="checkbox"
            value="guitar"
          />
        </div>
        <div
          class="checkbox"
        >
          <label
            class="label"
            for="voice"
          >
            voice
          </label>
          <input
            class="input"
            data-testid="voice"
            id="voice"
            name="instruments"
            type="checkbox"
            value="voice"
          />
        </div>
        <div
          class="checkbox"
        >
          <label
            class="label"
            for="drums"
          >
            drums
          </label>
          <input
            class="input"
            data-testid="drums"
            id="drums"
            name="instruments"
            type="checkbox"
            value="drums"
          />
        </div>
        <div
          class="checkbox"
        >
          <label
            class="label"
            for="violin"
          >
            violin
          </label>
          <input
            class="input"
            data-testid="violin"
            id="violin"
            name="instruments"
            type="checkbox"
            value="violin"
          />
        </div>
        <div
          class="checkbox"
        >
          <label
            class="label"
            for="other"
          >
            other
          </label>
          <input
            class="input"
            data-testid="other"
            id="other"
            name="instruments"
            type="checkbox"
            value="other"
          />
        </div>
      </div>
      <label
        class="label"
      >
         How did you hear about us?* 
      </label>
      <select
        class="leadSource"
        name="leadSource"
      >
        <option
          class="leadSourceOption"
          value=""
        >
          Please choose an option
        </option>
        <option
          class="leadSourceOption"
          data-testid="Referral"
          value="Referral"
        >
          Referral
        </option>
        <option
          class="leadSourceOption"
          data-testid="Yelp!"
          value="Yelp!"
        >
          Yelp!
        </option>
        <option
          class="leadSourceOption"
          data-testid="Google Search"
          value="Google Search"
        >
          Google Search
        </option>
        <option
          class="leadSourceOption"
          data-testid="Google Sponsored Ad"
          value="Google Sponsored Ad"
        >
          Google Sponsored Ad
        </option>
        <option
          class="leadSourceOption"
          data-testid="Facebook"
          value="Facebook"
        >
          Facebook
        </option>
        <option
          class="leadSourceOption"
          data-testid="Instagram"
          value="Instagram"
        >
          Instagram
        </option>
        <option
          class="leadSourceOption"
          data-testid="Other"
          value="Other"
        >
          Other
        </option>
      </select>
      <label
        class="label"
      >
         What is your availability for music lessons? 
      </label>
      <input
        class="input"
        name="availability"
        placeholder="Mon 3-6pm, Sat 8-10am"
        type="text"
      />
      <label
        class="label"
      >
         Additional info 
      </label>
      <textarea
        class="textarea"
        name="message"
        placeholder="Daughter is a beginner, 8 years old, likes Taylor Swift"
      />
      <button
        class="submit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
